applications/OptimizationApplication/python_scripts/utilities/response_function_implementor.py

```diff
@@ -0,0 +1,194 @@
+        pass
+
+    @abstractmethod
+    def GetStandardizedSensitivity(self, sensitivity_variable, sensitivity_data_container: ContainerVariableDataHolderUnion, solution_step_index: int = 0):
+        pass
+
+    @abstractmethod
+    def GetResponseType(self) -> str:
+        pass
+
+    @abstractmethod
+    def GetResponseInfo(self) -> str:
+        pass
+
+class ObjectiveResponseFunctionImplementor(ResponseFunctionImplementor):
+    def __init__(self, parameters: Kratos.Parameters, optimization_info: OptimizationInfo):
+        default_parameters = Kratos.Parameters("""{
+            "response_name": "",
+            "type"         : "",
+            "scaling"      : 1.0
+        }""")
+        parameters.ValidateAndAssignDefaults(default_parameters)
+
+        self.__scaling = parameters["scaling"].GetDouble()
+
+        self.__objective_type = parameters["type"].GetString()
+        if self.__objective_type == "minimization":
+            self.__scaling *= 1.0
+        elif self.__objective_type == "maximization":
+            self.__scaling *= -1.0
+        else:
+            raise RuntimeError(f"Requesting unsupported type {self.__objective_type} for objective response function. Supported types are: \n\tminimization\n\tmaximization")
+
+        super().__init__(parameters["response_name"].GetString(), optimization_info)
+
+    def GetStandardizedValue(self, solution_step_index: int = 0) -> float:
+        return self.GetValue(solution_step_index) * self.__scaling
+
+    def GetStandardizedSensitivity(self, sensitivity_variable, sensitivity_data_container: ContainerVariableDataHolderUnion, solution_step_index: int = 0):
+        self.GetSensitivity(sensitivity_variable, sensitivity_data_container, solution_step_index)
+        sensitivity_data_container *= self.__scaling
+
+    def GetResponseType(self) -> str:
+        return self.__objective_type
+
+    def GetResponseInfo(self) -> str:
+
+        if self._optimization_info["step"] > 1:
+            relative_change = (self.GetValue() / self.GetValue(1) - 1.0)
+        else:
+            relative_change = 0.0
+
+        absolute_change = (self.GetValue() / self._GetInitialResponseValue() - 1.0)
+
+        msg = "\tObjective info:"
+        msg += f"\n\t\t name          : {self.GetName()}"
+        msg += f"\n\t\t type          : {self.__objective_type}"
+        msg += f"\n\t\t value         : {self.GetValue()}"
+        msg += f"\n\t\t rel_change [%]: {relative_change * 100.0}"
+        msg += f"\n\t\t abs_change [%]: {absolute_change * 100.0}"
+
+        self._optimization_info.SetValue(f"{self._key_prefix}/relative_change", relative_change)
+        self._optimization_info.SetValue(f"{self._key_prefix}/absolute_change", absolute_change)
+        self._optimization_info.SetValue(f"{self._key_prefix}/type", self.__objective_type)
+        return msg
+
+class ConstraintResponseFunctionImplementor(ResponseFunctionImplementor):
```

Comment:
```suggestion
class Constraint(ResponseFunctionImplementor):
```

Move this to a different file.

Reply:
Do you mean the class is going to be in Constraint.py or so?

Reply:
yes, and done :)